activation/poetlistener.go

```diff
@@ -59,17 +59,24 @@ func (l *PoetListener) handlePoetProofMessage(poetProof service.GossipMessage) {
 		l.Log.Error("failed to unmarshal PoET membership proof: %v", err)
 		return
 	}
-	if processingIssue, err := l.poetDb.ValidateAndStorePoetProof(proofMessage.PoetProof, proofMessage.PoetId,
+	if processingIssue, err := l.poetDb.ValidatePoetProof(proofMessage.PoetProof, proofMessage.PoetId,
```

Comment:
the boolean here can be replaced if comparing error string to a "processing error" string

Reply:
I thought that having an explicit return value is clearer, but if we want to get rid of it, I don't like comparing strings as it's fragile (someone could change the string and not know that it breaks anything).

Another solution is to create a custom error type and check that. WDYT?